kirill-ianchuk/nodejs-aws-be
# Task9/bff service

### Main tasks
- [x] 3 - A working and correct express application should be in the bff-service folder. Reviewer can start this application locally with any valid configuration in the .env file and this application should work as described in the task 9.1

**Note**: The name of the service is a separate part of the URL, it is not one of the endpoints of product or cart service.
Examples:
GET http://localhost:3000/product-service/products
GET http://localhost:3000/cart-service/api/profile/cart

- [x] 5 - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service : product-service or cart-service. The response from the bff-service should be the same as if product-service or cart-service were called directly.

Endpoints:
http://kirill-ianchuk-bff-api-dev.eu-west-1.elasticbeanstalk.com/product-service/products
http://kirill-ianchuk-bff-api-dev.eu-west-1.elasticbeanstalk.com/cart-service/api/profile/cart

### Additional tasks

How to test:

1. Get products list:
http://kirill-ianchuk-bff-api-dev.eu-west-1.elasticbeanstalk.com/product-service/products
2. Create new product
POST http://kirill-ianchuk-bff-api-dev.eu-west-1.elasticbeanstalk.com/product-service/products
{
    "title": "ADD GUITAR TITLE",
    "description": "ADD GUITAR DESCRIPTION",
    "price": 1000,
    "count": 1
}
3. Get products list - result shouldn’t have the new product
4. Wait for more than 2 minutes
5. Get products list - result should have the new product

### Expected score: 7
